### ISSUE TITLE
[REVIEW]: qMRLab: Quantitative MRI Analysis, under one umbrella

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2343]
**Submitting author:** @agahkarakuzu (<a href="http://orcid.org/0000-0001-7283-271X">Agah Karakuzu</a>)
**Repository:** <a href="https://github.com/qMRLab/qMRLab" target ="_blank">https://github.com/qMRLab/qMRLab</a>
**Version:** v.2.4.1
**Editor:** @Kevin-Mattheus-Moerman
**Reviewer:** @grlee77, @mfroeling, @62442katieb
**Archive:** <a href="https://doi.org/10.5281/zenodo.4012104" target="_blank">10.5281/zenodo.4012104</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/2e87644fb8a9e93d48359565c7726f34/status.svg)](https://joss.theoj.org/papers/2e87644fb8a9e93d48359565c7726f34)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/2e87644fb8a9e93d48359565c7726f34"><img src="https://joss.theoj.org/papers/2e87644fb8a9e93d48359565c7726f34/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/2e87644fb8a9e93d48359565c7726f34/status.svg)](https://joss.theoj.org/papers/2e87644fb8a9e93d48359565c7726f34)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/qMRLab/qMRLab" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@grlee77 & @mfroeling & @62442katieb, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @Kevin-Mattheus-Moerman know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/qMRLab/qMRLab">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@agahkarakuzu) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/qMRLab/qMRLab">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@agahkarakuzu) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/qMRLab/qMRLab">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@agahkarakuzu) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @grlee77, @mfroeling, @62442katieb it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2343 with the following error: 

 sh: 0: getcwd() failed: No such file or directory
pandoc: 10.21105.joss.02343.pdf: openBinaryFile: does not exist (No such file or directory)
Looks like we failed to compile the PDF

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02343/joss.02343/10.21105.joss.02343.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- None

MISSING DOIs

- https://doi.org/10.21105/joss.00656 may be missing for title: QUIT: QUantitative imaging tools
- https://doi.org/10.1016/b978-012372560-8/50002-4 may be missing for title: Statistical parametric mapping
- https://doi.org/10.1101/343079 may be missing for title: Let’s talk about cardiac T1 mapping
- https://doi.org/10.1016/j.softx.2019.100369 may be missing for title: Total Mapping Toolbox (TOMATO): An open source library for cardiac magnetic resonance parametric mapping
- https://doi.org/10.5210/fm.v3i2.578 may be missing for title: The cathedral and the bazaar

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@grlee77, @mfroeling, @62442katieb this is where the review takes place. Please read the above instructions :point_up: (including suggestions on preventing unwanted GitHub notifications). 

You each have a set of checkboxes to guide you through the review process. It would be great if you could review this work in about 2 weeks. Although we can be flexible regarding review time (especially during these challenging times!!!) it would be great if you would let us know if you need more time. 

Thanks again for your help!!!! Let the reviewing begin :tada: :robot: :rocket:

**Comment by Kevin-Mattheus-Moerman** [comment]
@grlee77, @mfroeling, @62442katieb thanks again for your help with this review. I see @mfroeling has started. @grlee77, @62442katieb how are you getting on? Let me know if you have questions.

**Comment by grlee77** [comment]
Hey all, sorry about the delay. I will review this over the next day or so.

**Comment by grlee77** [comment]
@Kevin-Mattheus-Moerman, can you please resend the invitation. 

When I click the link to accept it in the first comment it says it is expired (so I am unable to edit the checklist).

**Comment by danielskatz** [comment]
@whedon re-invite @grlee77 as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@grlee77 please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by grlee77** [comment]
Re: **Contributions and authorship** item

Can the authors clarify briefly how authorship criteria were decided? I see that most authors overlap with the repository's significant source code contributors, but I understand contributions can also come in other forms. Two authors I don't see an obvious alias for in the commit history are Drs. Pike and Stikov. Also, there appear to be a few significant contributors (e.g. 30+ commits) who do not appear in the author list (although it is hard to tell for sure based on GitHub handles).

**Comment by grlee77** [comment]
Re: **installation instructions** item:

Your [documentation for this](https://qmrlab.readthedocs.io/en/master/install.html) is good, but it would be nice if you also listed which commercial Matlab toolboxes are required. I saw that for Octave, specific toolboxes were listed.

FWIW, I ran a few of the demos and then typed `license('inuse')` and saw
```
image_toolbox
matlab
optimization_toolbox
```
but I did not run everything, so there may be others as well.

**Comment by grlee77** [comment]
Please double-check the formatting of the references. For example the first reference (SPM) seems to be incomplete and the Weiskopf et. al. reference has a stray "Citeseer" listed.

**Comment by grlee77** [comment]
Re: **citations**:

The authors cite several relevant packages, but one area that does not seem to be represented is for Diffusion MRI. qMRLab does provide DTI, CHARMED and NODDI functionality, so it seems relevant to cite some packages in this area as well. Some common tools for that are free, but not open source, but a few open source ones I am aware of are:

[DIPY](https://dipy.org) and [DMIPY](https://github.com/AthenaEPI/dmipy) (Python)  (full disclosure: I have been an occasional DIPY contributor)
[Camino](http://camino.cs.ucl.ac.uk/) (Java)
[DSI Studio](http://dsi-studio.labsolver.org/) (C++)

Of course, it will not be possible to cite everything, but 1-2 sentences regarding other open-source diffusion processing tools would be nice.

**Comment by grlee77** [comment]
Regarding the included figure, it is pretty, but I wonder if there is any issue on JOSS's end to publish a figure including logos of other packages like Matlab, Docker, Plotly, etc. When we submitted the SciPy paper (albeit not for JOSS), there were originally various logos in a historical timeline figure, but the editorial team required the removal of any copyrighted logos prior to publication.

Also, when performing this review, I have only reviewed qMRLab itself, but did not evaluate the related projects like qMRFlow, etc. I think it is appropriate to mention these related projects in the text, but I don't think the figure needs to include them.  I guess what I am saying is that the figure looks a bit more like an advertisement for the broader software eco-system surrounding qMRLab and should perhaps be a bit more narrowly focused on the quantitative mapping subset described in the publication.

@Kevin-Mattheus-Moerman: can you advise on any potential issues regarding logos on JOSS's end?

**Comment by Kevin-Mattheus-Moerman** [comment]
> Regarding the included figure, it is pretty, but I wonder if there is any issue on JOSS's end to publish a figure including logos of other packages like Matlab, Docker, Plotly, etc. When we submitted the SciPy paper (albeit not for JOSS), there were originally various logos in a historical timeline figure, but the editorial team required the removal of any copyrighted logos prior to publication.
> 
> Also, when performing this review, I have only reviewed qMRLab itself, but did not evaluate the related projects like qMRFlow, etc. I think it is appropriate to mention these related projects in the text, but I don't think the figure needs to include them. I guess what I am saying is that the figure looks a bit more like an advertisement for the broader software eco-system surrounding qMRLab and should perhaps be a bit more narrowly focused on the quantitative mapping subset described in the publication.
> 
> @Kevin-Mattheus-Moerman: can you advise on any potential issues regarding logos on JOSS's end?

@grlee77 thanks for raising this issue with us. The logos are acceptable.

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu can you respond to the comments by @grlee77 ?

**Comment by 62442katieb** [comment]
> Re: **Contributions and authorship** item
> 
> Can the authors clarify briefly how authorship criteria were decided? I see that most authors overlap with the repository's significant source code contributors, but I understand contributions can also come in other forms. Two authors I don't see an obvious alias for in the commit history are Drs. Pike and Stikov. Also, there appear to be a few significant contributors (e.g. 30+ commits) who do not appear in the author list (although it is hard to tell for sure based on GitHub handles).

I would also like clarification on the authorship decisions. Thanks!

**Comment by agahkarakuzu** [comment]
@grlee77 @62442katieb 

Re: Contributions and authorship item

> Two authors I don't see an obvious alias for in the commit history are Drs. Pike and Stikov.

* Bruce Pike and Nikola Stikov (my advisor) are principal investigators of the research labs funding our project

> Can the authors clarify briefly how authorship criteria were decided? 

* For the authorship, we thought maintainers, active contributors and PIs fit for the authorship. These are the contributors regularly joining our bi-weekly meetings. For clarity,  below is a brief breakdown of authorship roles: 

     * Agah Karakuzu: Maintainer, improved the qMRLab framework, significant contributions to the codebase, added DevOps and new qMRI models, wrote the manuscript. 
     * Mathieu Boudreau: Maintainer, significant contributions to the codebase, added new qMRI models, revised the manuscript
     * Tanguy Duval: Former maintainer, created qMRLab framework based on qMTLab, significant contributions to the codebase, revised the manuscript
     * Tommy Boshkovski: Active contributor, added qMRI models, helped improve documentation, created the web portal, revised the manuscript. 
     * Ilana R. Leppert: Active contributor, added qMRI models, revised the manuscript. 
     * Bruce Pike: Principal Investigator, contributed funds for the development of the software, revised the manuscript. 
     * Julien Cohen-Adad: Principal Investigator, contributions to the codebase,  contributed funds for the development of the software, provided valuable insights on software development, revised the manuscript. 
     * Nikola Stikov: Principal Investigator, main supervisor of the project and academic supervisor of A.K. and T.B., contributed funds for the development of the software, revised the manuscript 

> Also, there appear to be a few significant contributors (e.g. 30+ commits) who do not appear in the author list (although it is hard to tell for sure based on GitHub handles).

* I would like to note that later on in the project, we started using `squash and merge` option to merge the commits into the `master`. Therefore, at least for this codebase, number of commits is not necessarily an objective measure of the relative code contributions. Nonetheless, we are more than happy to welcome all the contributors as authors if the JOSS policy (@Kevin-Mattheus-Moerman ) requires and/or the reviewers @grlee77 @62442katieb strongly recommend to do so. 

> Also, when performing this review, I have only reviewed qMRLab itself, but did not evaluate the related projects like qMRFlow, etc. I think it is appropriate to mention these related projects in the text, but I don't think the figure needs to include them.

Thank you @grlee77, I agree with your point. I will update the figure to confine the content to qMRLab codebase only and simplify it. 

- [ ] The comment will be addressed in the manuscript.

> The authors cite several relevant packages, but one area that does not seem to be represented is for Diffusion MRI. qMRLab does provide DTI, CHARMED and NODDI functionality, so it seems relevant to cite some packages in this area as well.

Thank you for the suggestion. I agree that including some open source examples from the Diffusion MRI field is relevant.

- [ ] The comment will be addressed in the manuscript.

- [ ]  Please double-check the formatting of the references.

- [ ]  it would be nice if you also listed which commercial Matlab toolboxes are required

Thank you so much @grlee77 for your valuable comments, I will soon address them and make a PR.

**Comment by agahkarakuzu** [comment]
@Kevin-Mattheus-Moerman I responded to the comments by @grlee77 and @62442katieb and addressed the issues raised in [qMRLab PR #399](https://github.com/qMRLab/qMRLab/pull/399) ([`joss_rev1`](https://github.com/qMRLab/qMRLab/pull/399) into `master`).

**Comment by grlee77** [comment]
> I would like to note that later on in the project, we started using squash and merge option to merge the commits into the master. Therefore, at least for this codebase, number of commits is not necessarily an objective measure of the relative code contributions. Nonetheless, we are more than happy to welcome all the contributors as authors if the JOSS policy (@Kevin-Mattheus-Moerman ) requires and/or the reviewers @grlee77 @62442katieb strongly recommend to do so.

Agreed, Re: commits not being an ideal measure of effort. I'm not sure in regards to JOSS policy, but given that it only seems to be a few additional contributors, it shouldn't be much work to invite them. I prefer to lean toward the side of being more inclusive rather than less. If they are uninterested or you do not receive a timely response, then do not include them among the primary authors, perhaps just adding a generic acknowledgement of the contributions of "qMRLab community contributors".

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu @grlee77 
In relation to authorship here is our policy on this: https://joss.readthedocs.io/en/latest/submitting.html#authorship. In short _"The authors themselves assume responsibility for deciding who should be credited with co-authorship, and co-authors must always agree to be listed."_, 
In this case some contributors were identified which perhaps could be added as authors. I leave this decision with @agahkarakuzu. If this concerns a small amount of contributors, and you do not mind, then I do suggest inviting them (if their contributions are sufficient enough) to see if they feel they would like to be added as a co-author here.

**Comment by Kevin-Mattheus-Moerman** [comment]
@mfroeling any updates on your end? How are you getting on with this review? Thanks.

**Comment by agahkarakuzu** [comment]
@Kevin-Mattheus-Moerman thanks for the clarification re authorship, I will discuss this with the team and come up with a time efficient strategy.

**Comment by grlee77** [comment]
It looks like qMRLab/qMRLab#399 addresses all of my other comments (the new figure looks good!). Thanks @agahkarakuzu for the quick update.

**Comment by agahkarakuzu** [comment]
@grlee77 regarding authorship, we will send all the contributors an email and ask if they would like to be co-author. We will wait for a week to collect the responses and the necessary information, then update the author list.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- None

MISSING DOIs

- https://doi.org/10.21105/joss.00656 may be missing for title: QUIT: QUantitative imaging tools
- https://doi.org/10.1016/b978-012372560-8/50002-4 may be missing for title: Statistical parametric mapping
- https://doi.org/10.1101/343079 may be missing for title: Let’s talk about cardiac T1 mapping
- https://doi.org/10.1016/j.softx.2019.100369 may be missing for title: Total Mapping Toolbox (TOMATO): An open source library for cardiac magnetic resonance parametric mapping
- https://doi.org/10.4135/9781412950657.n33 may be missing for title: The cathedral and the bazaar

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu can you check those missing DOI's? :point_up:

**Comment by Kevin-Mattheus-Moerman** [comment]
@grlee77 just had a look at your unchecked boxes. "References" and "Contribution/authorship" are being dealt with. Can you summarize why "State of the field" is unchecked? Thanks!

**Comment by Kevin-Mattheus-Moerman** [comment]
@mfroeling how are you getting on with this review? Let me know if you have any questions.

**Comment by grlee77** [comment]
> @grlee77 just had a look at your unchecked boxes. "References" and "Contribution/authorship" are being dealt with. Can you summarize why "State of the field" is unchecked? Thanks!

State of the field was also addressed in their PR (discussion of existing diffusion-related software was added). I wasn't sure if I should wait until it was merged to check them off, but have gone ahead and checked them now so I don't hold up the review.

**Comment by agahkarakuzu** [comment]
@grlee77 I sent contributors an email and waited for a week for their response. Now I will add those accepted co-authorship in the PR and then merge.

**Comment by agahkarakuzu** [comment]
@Kevin-Mattheus-Moerman I added missing dois manually (added `doi` field(s) in `paper.bib`), I checked them one by one and they all resolve correctly. Now they appear in the refs:

![image](https://user-images.githubusercontent.com/9632322/87534488-16d04a80-c664-11ea-9858-6c277f488d62.png)

If this looks OK, I will merge the PR to master.

**Comment by Kevin-Mattheus-Moerman** [comment]
Looks good. Perhaps merge and run `@whedon generate pdf` and `@whedon check references` here.

**Comment by agahkarakuzu** [comment]
@whedon generate pdf

**Comment by agahkarakuzu** [comment]
@whedon check references

**Comment by whedon** [comment]
PDF failed to compile for issue #2343 with the following error: 

 Could not find bibliography file: paper.bib
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.21105/joss.00656 is OK
- 10.1016/b978-012372560-8/50002-4 is OK
- 10.1101/343079 is OK
- 10.1016/j.softx.2019.100369 is OK
- 10.4135/9781412950657.n33 is OK

MISSING DOIs

- https://doi.org/10.1038/nbt.3820 may be missing for title: Nextflow enables reproducible computational workflows
- https://doi.org/10.1101/631952 may be missing for title: TractoFlow: A robust, efficient and reproducible diffusion MRI pipeline leveraging Nextflow & Singularity
- https://doi.org/10.3389/fninf.2014.00008 may be missing for title: Dipy, a library for the analysis of diffusion MRI data
- https://doi.org/10.3389/fninf.2019.00064 may be missing for title: The dmipy toolbox: Diffusion mri multi-compartment modeling and microstructure recovery made easy

INVALID DOIs

- None
```

**Comment by mfroeling** [comment]
@agahkarakuzu My appologies for the delay. I was a bit bussy and needed to find a matlab version with the correct toolboxes which in our IT department had a bit of a delay. 

some issues that I think need to be solved.

- [ ] When the toolbox starts it needs a check if the needed Matlab dependencies are installed. The first few tries I had issues during data processing that were due to missing or badly installed packages. Especially for the missing packages, there should be a strong warning when starting the toolbox that it will not run properly.

- [ ] Example data is provided for the T1 relaxometry in the documentation. After some browsing the GUI I found the download example button also for other modalities. This should be better documented. I started with the command line and there I could not find a list of available example data.

- [ ] Each model needs a specific data setup that is apparently inferred from the files in the folder. One can load each data needed separately or by selecting the containing folder. However, for me, it is not clear how the data structure/naming should be for every type of data supported if one wants to used the predefined data format. For example, when additional data quantitative data is loaded how should this be stored (e.g. B1 can be fraction or percent).

- [ ] I loaded one of my own diffusion sets and the bval/bvec files were not loaded or asked for. without this information the tool still allowed me to click the fit button resulting in errors. Only after some exploration I found the load bval/bvec in the options pannel. Nowhere in the help of the diffusion does it state that I have to load my own bval/bvec files.

- [ ] different models ask for different input files. This is nicely provided in the GUI however I would be nice if mandatory and optional inputs would be clear. For example, with DTI processing I don't need a mask or SigmaNoise to be able to fit a Tensor.

- [ ] When I loaded the example diffusion data and then changed the input to one of my own datasets the mask of the previous dataset remained in the Mask input field only after clicking browse and cancel is was cleared.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02343/joss.02343/10.21105.joss.02343.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.21105/joss.00656 is OK
- 10.1016/b978-012372560-8/50002-4 is OK
- 10.1101/343079 is OK
- 10.1016/j.softx.2019.100369 is OK
- 10.4135/9781412950657.n33 is OK

MISSING DOIs

- https://doi.org/10.1038/nbt.3820 may be missing for title: Nextflow enables reproducible computational workflows
- https://doi.org/10.1101/631952 may be missing for title: TractoFlow: A robust, efficient and reproducible diffusion MRI pipeline leveraging Nextflow & Singularity
- https://doi.org/10.3389/fninf.2014.00008 may be missing for title: Dipy, a library for the analysis of diffusion MRI data
- https://doi.org/10.3389/fninf.2019.00064 may be missing for title: The dmipy toolbox: Diffusion mri multi-compartment modeling and microstructure recovery made easy

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu your paper compiled now but it looks like some DOI's are still missing :point_up: can you check/add those?

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu :wave: how are you getting on? There are some comments to work on above from myself and the reviewers.

**Comment by agahkarakuzu** [comment]
@Kevin-Mattheus-Moerman I am working on a branch, I will address all the comments this week.

**Comment by agahkarakuzu** [comment]
- [x] Add missing DOIs (@Kevin-Mattheus-Moerman)
 
Comments by @mfroeling 
- [x]  When the toolbox starts it needs a check if the needed Matlab dependencies are installed. The first few tries I had issues during data processing that were due to missing or badly installed packages. Especially for the missing packages, there should be a strong warning when starting the toolbox that it will not run properly.

- [x]  Example data is provided for the T1 relaxometry in the documentation. After some browsing the GUI I found the download example button also for other modalities. This should be better documented. I started with the command line and there I could not find a list of available example data.

- [x]  Each model needs a specific data setup that is apparently inferred from the files in the folder. One can load each data needed separately or by selecting the containing folder. However, for me, it is not clear how the data structure/naming should be for every type of data supported if one wants to used the predefined data format. For example, when additional data quantitative data is loaded how should this be stored (e.g. B1 can be fraction or percent).

- [x]  I loaded one of my own diffusion sets and the bval/bvec files were not loaded or asked for. without this information the tool still allowed me to click the fit button resulting in errors. Only after some exploration I found the load bval/bvec in the options pannel. Nowhere in the help of the diffusion does it state that I have to load my own bval/bvec files.

- [ ]  different models ask for different input files. This is nicely provided in the GUI however I would be nice if mandatory and optional inputs would be clear. For example, with DTI processing I don't need a mask or SigmaNoise to be able to fit a Tensor.

I will open a PR on `qMRLab/qMRLab` repo (`joss_rev2`-->`master`) to respond to the comments in this issue once I check all the items above. Thank you so much, @mfroeling!

**Comment by mfroeling** [comment]
@agahkarakuzu Great i will look at the updates asap.

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu can you update on progress? Have the DOI issues been resolved? Have you commented on/resolved all reviewer issues? Feel free to regenerate the pdf (`@whedon generate pdf`) here and also to check for references (`@whedon check references`).

**Comment by agahkarakuzu** [comment]
@Kevin-Mattheus-Moerman I resolved all the reviewer issues, just pushed a bug fix, waiting for CI tests, then going to merge `joss_rev2` branch into the `master` today and follow your instructions.

**Comment by agahkarakuzu** [comment]
@whedon generate pdf

**Comment by agahkarakuzu** [comment]
@whedon check references

**Comment by whedon** [comment]
PDF failed to compile for issue #2343 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-0e09ec0a48e3/lib/whedon/compilers.rb:158:in `initialize': No such file or directory @ rb_sysopen - tmp/2343/markdown-metadata.yaml (Errno::ENOENT)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-0e09ec0a48e3/lib/whedon/compilers.rb:158:in `open'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-0e09ec0a48e3/lib/whedon/compilers.rb:158:in `pdf_from_markdown'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-0e09ec0a48e3/lib/whedon/compilers.rb:13:in `generate_pdf'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-0e09ec0a48e3/bin/whedon:56:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-0e09ec0a48e3/bin/whedon:119:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.21105/joss.00656 is OK
- 10.1016/b978-012372560-8/50002-4 is OK
- 10.1101/343079 is OK
- 10.1016/j.softx.2019.100369 is OK
- 10.4135/9781412950657.n33 is OK
- 10.1038/nbt.3820 is OK
- oi.org/10.1101/631952 is OK
- doi.org/10.3389/fninf.2014.00008 is OK
- doi.org/10.3389/fninf.2019.00064 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by Kevin-Mattheus-Moerman** [comment]
@62442katieb thanks again for your help here. You have some boxes that remain unticked. Can you summarize what the authors can do to resolve any remaining issues you have? Thanks!

**Comment by 62442katieb** [comment]
> @62442katieb thanks again for your help here. You have some boxes that remain unticked. Can you summarize what the authors can do to resolve any remaining issues you have? Thanks!

Hi @Kevin-Mattheus-Moerman ! I've had a couple issues with installation on our cluster. Those issues have been resolved, but now it appears the download links in the [installation instructions](https://qmrlab.readthedocs.io/en/master/install.html) are broken (https://github.com/qMRLab/qMRLab/releases/latest/download/qMRLab_latest.zip, https://github.com/qMRLab/qMRLab/releases/download/v2.4.0/qMRLab_latest.zip). 

Please let me know if there is another location from which I should be downloading. Thanks!

**Comment by agahkarakuzu** [comment]
@62442katieb can you download the `v2.4.0` from the assets section of [this link](https://github.com/qMRLab/qMRLab/releases/tag/v2.4.0)? 

You can also check out the [master](https://github.com/qMRLab/qMRLab/archive/master.zip) version, where previous comments were addressed.

I'll fix the broken hyperlinks.

**Comment by Kevin-Mattheus-Moerman** [comment]
@62442katieb I see @agahkarakuzu was able to make changes to address some of your points. Please let us know if there are any other remaining issues. Thanks again!

**Comment by Kevin-Mattheus-Moerman** [comment]
@62442katieb :point_up: let us know if all issues have not been resolved or if you have further comments. Thanks.

**Comment by 62442katieb** [comment]
Hi @Kevin-Mattheus-Moerman, I apologize for the delay. All issues have been resolved. Thanks!

**Comment by Kevin-Mattheus-Moerman** [comment]
That is fine. Thanks for your help here @62442katieb

**Comment by Kevin-Mattheus-Moerman** [comment]
@62442katieb would you mind ticking all boxes :point_up: if all issues have been resolved and if you recommend acceptance? Thanks

**Comment by Kevin-Mattheus-Moerman** [comment]
@62442katieb thanks

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02343/joss.02343/10.21105.joss.02343.pdf)

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu I've reviewed your paper and it seems fine. Below are the last steps to complete: 

- [ ] Can you archive a copy of the software on ZENODO and report back here with the DOI of the archived version (do not alter or set the DOI yourself, please accept the ZENODO generated DOI)? Please ensure that the title and author list is the same as the JOSS paper for this ZENODO archived version (you may need to edit this meta-data manually). 

- [ ] Can you inform me about the version tag for this work. Is it still at `v2.4.0` or has it moved on? 

- [ ] Your paper is about to be accepted in JOSS. I recommend that you review the paper yourself once more, and that you also check the author list, affiliations, and the acknowledgements section for accuracy. (call `@whedon generate pdf` here to update the paper).

**Comment by agahkarakuzu** [comment]
@Kevin-Mattheus-Moerman thanks! Now that the review has been completed, I would like to release `v.2.4.1` that includes changes suggested by the reviewers and update the citation information to the JOSS reference. Is there a way to know how the citation is going to look like at this stage?

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu That is fine, I can set `v.2.4.1` as the version once you confirm that. That way it will be recorded with the paper publication meta data.

@openjournals/dev @arfon I take it that the DOI for this submission will become `https://doi.org/10.21105/joss.02343` (as currently shown on the proof)? Perhaps @agahkarakuzu can use that with the citation information for this release.

**Comment by arfon** [comment]
Yes, `https://doi.org/10.21105/joss.02343` will be the DOI.

**Comment by agahkarakuzu** [comment]
Thank you @arfon and @Kevin-Mattheus-Moerman I will make the release and go through the checklist.

**Comment by agahkarakuzu** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02343/joss.02343/10.21105.joss.02343.pdf)

**Comment by agahkarakuzu** [comment]
@Kevin-Mattheus-Moerman 

> Can you archive a copy of the software on ZENODO and report back here with the DOI of the archived version (do not alter or set the DOI yourself, please accept the ZENODO generated DOI)? Please ensure that the title and author list is the same as the JOSS paper for this ZENODO archived version (you may need to edit this meta-data manually).

**Zenodo DOI:** 10.5281/zenodo.4012104

[![DOI](https://zenodo.org/badge/41506224.svg)](https://zenodo.org/badge/latestdoi/41506224)

✅ Author list on Zenodo matches that on the PDF. 

>  Can you inform me about the version tag for this work. Is it still at v2.4.0 or has it moved on?

✅ Tag is now at [`v2.4.1`](https://github.com/qMRLab/qMRLab/releases/tag/v2.4.1). 

The [`v2.4.1.` release pipeline](https://dev.azure.com/neuropoly/qMRLab/_releaseProgress?_a=release-pipeline-progress&releaseId=56) has been succeeded. Compiled (MCR) version of the `v2.4.1` is pushed to [DockerHub](https://hub.docker.com/u/qmrlab) along with other virtual environments. 

> Your paper is about to be accepted in JOSS. I recommend that you review the paper yourself once more, and that you also check the author list, affiliations, and the acknowledgements section for accuracy.

✅ Author list, affiliations and the acknowledgements are OK.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set v.2.4.1 as version

**Comment by whedon** [comment]
OK. v.2.4.1 is the version.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon set 10.5281/zenodo.4012104 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4012104" target="_blank">10.5281/zenodo.4012104</a> is the archive.

**Comment by Kevin-Mattheus-Moerman** [comment]
@agahkarakuzu can you amend the title of the ZENODO archived version to match that of the paper? It is currently `qMRLab/qMRLab: Release v2.4.1 Zenodo Archive` but should be `qMRLab: Quantitative MRI analysis, under one umbrella`

**Comment by agahkarakuzu** [comment]
@Kevin-Mattheus-Moerman I fixed the title.

**Comment by Kevin-Mattheus-Moerman** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1699

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1699, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
Hi @agahkarakuzu, I am the rotating EIC and I just read through your paper and I see that the capitalization in your references is incorrect. I made changes to your .bib file in [this PR](https://github.com/qMRLab/qMRLab/pull/407).

**Comment by kthyng** [comment]
@whedon generate pdf

**Comment by kthyng** [comment]
Ok it looks good to me!

**Comment by kthyng** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.21105/joss.00656 is OK
- 10.1016/b978-012372560-8/50002-4 is OK
- 10.1101/343079 is OK
- 10.1016/j.softx.2019.100369 is OK
- 10.4135/9781412950657.n33 is OK
- 10.1038/nbt.3820 is OK
- oi.org/10.1101/631952 is OK
- doi.org/10.3389/fninf.2014.00008 is OK
- doi.org/10.3389/fninf.2019.00064 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by kthyng** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1701

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1701, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1301507377955561475) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1702
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02343](https://doi.org/10.21105/joss.02343)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats to @agahkarakuzu and @mathieuboudreau on your new publication!! Thanks to editor @Kevin-Mattheus-Moerman
 and reviewers @grlee77, @mfroeling, and @62442katieb. We wouldn't be able to do this without your time and expertise.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02343/status.svg)](https://doi.org/10.21105/joss.02343)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02343">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02343/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02343/status.svg
   :target: https://doi.org/10.21105/joss.02343
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02343/status.svg)](https://doi.org/10.21105/joss.02343)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
